cmd/login/runner.go

```diff
@@ -107,27 +108,43 @@ func (r *runner) run(ctx context.Context, cmd *cobra.Command, args []string) err
 	return nil
 }
 
-func (r *runner) tryToGetCurrentContext(ctx context.Context) (string, error) {
+func (r *runner) tryToGetCurrentContexts(ctx context.Context) (string, string, error) {
 	config, err := r.commonConfig.GetConfigAccess().GetStartingConfig()
 	if err != nil {
-		return "", microerror.Mask(err)
+		return "", "", microerror.Mask(err)
 	}
-	return config.CurrentContext, nil
+	contextOverride := r.commonConfig.GetContextOverride()
+	if _, ok := config.Contexts[contextOverride]; !ok {
+		contextOverride = ""
+	}
+	return config.CurrentContext, contextOverride, nil
 }
 
 func (r *runner) setLoginOptions(ctx context.Context, args *[]string) {
-	originContext, err := r.tryToGetCurrentContext(ctx)
+	originContext, contextOverride, err := r.tryToGetCurrentContexts(ctx)
 	if err != nil {
 		fmt.Fprintln(r.stdout, color.YellowString("Failed trying to determine current context. %s", err))
 	}
+
+	hasWCNameFlag := r.flag.WCName != ""
+	hasSelfContainedFlag := r.flag.SelfContained != ""
+	hasContextOverride := contextOverride != ""
+
+	// indicates whether it is desired to update current context in the kubeconfig file
+	shouldSwitchContextInConfig := !hasContextOverride && (hasWCNameFlag || !(hasSelfContainedFlag || r.flag.KeepContext))
+
+	// indicates whether it is desired to update current context in the kubeconfig file to the wc client context
+	shouldSwitchContentInClusterConfig := hasWCNameFlag && !(hasSelfContainedFlag || r.flag.KeepContext)
+
```

Comment:
Structured the conditions differently and tried to use more descriptive names to make them self-explanatory.